stl/inc/xpolymorphic_allocator.h

```diff
@@ -242,36 +261,39 @@ namespace pmr {
         template <class _Uty, class... _Types>
         _NODISCARD_RAW_PTR_ALLOC __declspec(allocator) _Uty* new_object(_Types&&... _Args) {
             _Uty* const _Ptr = allocate_object<_Uty>();
-            _TRY_BEGIN
+
+            _Deallocate_bytes_guard _Guard{_Resource, _Ptr, sizeof(_Uty), alignof(_Uty)};

```

Comment:
Instead of capturing the size and alignment at runtime, we could simply template the guard to bake this in.